src/main/java/net/sf/log4jdbc/StatementSpy.java

```diff
@@ -25,15 +25,34 @@
 import java.util.List;
 import java.util.ArrayList;
 
+import static net.sf.log4jdbc.Utilities.getConfigProp;
+
 /**
  * Wraps a Statement and reports method calls, returns and exceptions.
  *
  * jdbc 4 version
  *
  * @author Arthur Blake
  */
-public class StatementSpy implements Statement, Spy
-{
+public class StatementSpy implements Statement, Spy {
+  static final String SQL_SHOW_PARAMS_KEY = "LOG4JDBC_SQL_SHOW_PARAMS";
+
+  static final boolean showParams;
+
+  static {
+    showParams = getConfigProp(SQL_SHOW_PARAMS_KEY, v -> {

```

Comment:
style nit:

```java
public static void canShowSqlValues() {
  return getConfigProp(SQL_SHOW_PARAMS_KEY, ...)
}
```

and then in the class

```java
static final boolean showParams = canShowSqlValues()
```

**`[non-blocking]`** **`[soft-suggestion]`**

Reply:
It's a matter of taste and I have no preferences here. But I see public `canShowSqlValues` potentially unsafe because it reads system properties and system properties are mutable, so it is possible to get different results on different invocations of `canShowSqlValues`. So I'd rather make it private.

Reply:
Right .. I did not intend to use `public`. It should be private. Am trying to apply RAII outside of the clunky Java syntax.

I will leave it to you if you would like to make the change. Not a blocker.